File: apps/workspace-engine/pkg/reconcile/events/jobdispatch.go
package events

import (
	"context"
	"workspace-engine/pkg/reconcile"
)

const JobDispatchKind = "job-dispatch"

type JobDispatchParams struct {
	WorkspaceID string
	JobID       string
}

func EnqueueJobDispatch(queue reconcile.Queue, ctx context.Context, params JobDispatchParams) error {
	return queue.Enqueue(ctx, reconcile.EnqueueParams{
		WorkspaceID: params.WorkspaceID,
		Kind:        JobDispatchKind,
		ScopeType:   "job",
		ScopeID:     params.JobID,
	})
}

File: packages/db/src/reconcilers/job-dispatch.ts
import type { Tx } from "../common";
import { enqueue } from "./enqueue.js";

const JOB_DISPATCH_KIND = "job-dispatch";

export const enqueueJobDispatch = async (
  db: Tx,
  params: { workspaceId: string; jobId: string },
) =>
  enqueue(db, {
    workspaceId: params.workspaceId,
    kind: JOB_DISPATCH_KIND,
    scopeType: "job",
    scopeId: params.jobId,
  });
